Lib/email/utils.py

```diff
@@ -103,12 +104,126 @@ def formataddr(pair, charset='utf-8'):
     return address
 
 
+def _iter_escaped_chars(addr):
```

Comment:
It can use regular expressions:
```py
for m in re.finditer(r'(?s)\\?.', addr):
    yield (m.begin(), m.group())
```
Perhaps functions that use `_iter_escaped_chars()` can also be written using efficient regular expressions to avoid handling ever char separately. For example, `_strip_quoted_realnames()` could use `r'(?s)(?:\\.|[^"])*|(")'` or something like.

But do what looks good to you, I can optimize it later.

Reply:
> Perhaps functions that use _iter_escaped_chars() can also be written using efficient regular expressions to avoid handling ever char separately.

This PR is based on PR #108250 which uses the following code to strip quoted real names:

```py

    # When a comma is used in the Real Name part it is not a deliminator
    # So strip those out before counting the commas
    pattern = r'"[^"]*,[^"]*"'
    ...
    for v in fieldvalues:
        v = re.sub(pattern, '', v)
        ...
```

But this regular expression doesn't handle escape quote character (`\"`). So I wrote `_strip_quoted_realnames()` to replace this invalid regex.

I'm not comfortable with regex to write a parser when the grammar is that complicated. For example, antislash can also be escaped: ``\\``. I didn't check RFC carefully. I mostly tried to support escaped quote character (`\"`).

Also, my priority here is more to get a fix as soon as possible, since the vulnerability was reported in March, and I have to fix the CVE at work as soon as possible.